pkg/operator/saptuneapplysolution_v1.go

```diff
@@ -76,9 +84,16 @@ type saptuneOperationDiffOutput struct {
 	Solution string `json:"solution"`
 }
 
-func WithSaptuneClient(saptuneClient saptune.Saptune) SaptuneApplySolutionOption {
-	return func(o *SaptuneApplySolution) {
-		o.saptune = saptuneClient
+func WithSaptuneClient[T SaptuneOperator](saptuneClient saptune.Saptune) Option[T] {
```

Comment:
I am not sure this is the best way to do it, but I would need to look deeper into go generics 🤷

Reply:
I as well found this solution, and honestly didn't like it hehe
I vote for having duplicated code.
`WithSaptuneClientChange` and `WithSaptuneClientApply`.
But well, just an opinion

Reply:
I am not gonna argue now so here's the change 16ff1080a7afd713600cd1288aa44d3f92cf81b4

We will need to face the duplications that might rise. Let's see if they remain manageable.

For my personal taste we already passed the threshold 🙈, however let's observe how it evolves.